src/app/room/components/room.component.ts

```diff
@@ -44,15 +44,28 @@ export class RoomComponent implements OnInit {
 
   ngOnInit(): void {
     this.loadingStatus.isLoading = true;
+    const roomId = this.route.snapshot.params[constants.routeParams.id];
     this.currentPlayer$.pipe(take(1)).subscribe(player => {
       if (!player.id) {
-        this.playerCreationService.createPlayer(this.route.snapshot.params[constants.routeParams.id]).subscribe(_ => this.loadingStatus.isLoading = false);
+        this.createNewPlayer(roomId);
       } else {
-        this.loadingStatus.isLoading = false;
+        this.resetExistingPlayer(roomId);
       }
     });
   }
 
+  private createNewPlayer(roomId: string): void {
+    this.playerCreationService.createPlayer(roomId).subscribe(_ => this.loadingStatus.isLoading = false);
+  }
+
+  private resetExistingPlayer(roomId: string): void {
+    this.firebasePlayerService.updateCurrent({
+      room: roomId,
+      isObserver: false,
+      choice: null,
+    } as Partial<Player>).subscribe(() => this.loadingStatus = { isLoading: false });
```

Comment:
Can you add `this.loadingStatus = { isLoading: false }` to `finalize` in the parent method, please?

Reply:
Refactored it all a little, still not perfect but much better I think, check it out